tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -303,8 +304,15 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 		}
 
 		ginkgo.By(fmt.Sprintf("Adopt orphaned service created by helm"))
-		tc.Spec.TiDB.Service = &v1alpha1.TiDBServiceSpec{}
-		_, err = cli.PingcapV1alpha1().TidbClusters(ns).Update(tc)
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			tc, err := cli.PingcapV1alpha1().TidbClusters(ns).Get(tcName, metav1.GetOptions{})
```

Comment:
IIRC, the local variable `tc` will shadow the `tc` in the outer scope, the outer `tc` still remains un-updated and differs the actual `tc` spec in the cluster, which may  cause later mistake